pkg/operator/controller/controller.go

```diff
@@ -247,6 +249,19 @@ func (r *reconciler) enforceEffectiveHighAvailability(ci *ingressv1alpha1.Cluste
 	return nil
 }
 
+// ensureClusterIngressFinalized adds ClusterIngressFinalizer to ingress
+// if it does not exist and does not contain a DeletionTimestamp.
+func (r *reconciler) ensureClusterIngressFinalized(ingress *ingressv1alpha1.ClusterIngress) error {
+	if ingress.ObjectMeta.DeletionTimestamp.IsZero() && !slice.ContainsString(ingress.Finalizers, ClusterIngressFinalizer) || ingress.Finalizers == nil {
```

Comment:
If following the `enforce...` pattern above I don't think the deletion check is necessary

Reply:
Disregard (
https://github.com/openshift/cluster-ingress-operator/pull/146#discussion_r261014348). Only suggestion here now is a potential rename to "enforceFinalizer" for consistency

Reply:
I don't understand how https://github.com/openshift/cluster-ingress-operator/pull/146#discussion_r261014348 means the deletion check is needed.  Anyway, the logic looks wrong and might benefit from reordering, parentheses, or simplification: Do we really want to add the finalizer if `ingress.Finalizers == nil`, even if the ingress is deleted? Do we need the `ingress.Finalizers == nil` check at all (since `slice.ContainsString` treats a nil slice as an empty slice)?

Reply:
I'll update the method based on your feedback.